### ISSUE TITLE
Time series "staleness" issue with the script_exporter "ndt_e2e" probe

### ISSUE THREAD

**Issue by nkinkade** [m-lab/prometheus-support#186]
@stephen-soltesz:

I've been trying to debug an issue with determining when NDT is up or down. I would expect this Prom query to return almost every node:

```
  probe_success{service="ndt_ssl"} AND ON(machine)
      up{service="nodeexporter"} == 1 UNLESS ON(machine)
      lame_duck_node{} == 1
  AND ON(machine) (
    probe_success{service="ndt_ssl"} == 1 AND ON(machine)
    probe_success{service="ndt_raw"} == 1 AND ON(machine)
    script_success{service="ndt_e2e"}  == 1 AND ON(machine)
    vdlimit_used{experiment="ndt.iupui"} / vdlimit_total{experiment="ndt.iupui"} < 0.8
  )
```

But in practice it only returns around half of all nodes. Inspecting the problem, I found that the issue is likely due to the fact that the `ndt_e2e` check has a `scrape_interval` of 10m, while everything else has a 1m interval. I think we are running up against the <v2.0 Prom issue of "[staleness](https://prometheus.io/docs/prometheus/latest/querying/basics/#staleness)". In versions less than 2.0 a vector is "stale" if there is no value more recent than 5m. It would appear that in [versions >=2.0 this limitation is removed](https://www.slideshare.net/brianbrazil/staleness-and-isolation-in-prometheus-20-promcon-2017).

I haven't yet figured out any reasonable workaround for our [current version of Prometheus](https://github.com/m-lab/prometheus-support/blob/master/k8s/prometheus-federation/deployments/prometheus.yml#L49) (1.8.3).

Questions: how can this be worked around? If not, then is it worth considering an upgrade to Prom?

**Comment by stephen-soltesz** [comment]
Some options:

* we could work on the upgrade to 2.0 -- we should talk about that at the next prometheus sync either way -- this will take some time.
* change the staleness flag to 10m or 11m to match the frequency of e2e tests.
* run the e2e test more frequently.

Have I missed any?

**Comment by stephen-soltesz** [comment]
It sounds like prometheus does not behave well when query-staleness-delta is much larger than the default and the recommended scrape interval is no more than 2m. But, we want to run ndt e2e checks on the order of every ~10m.

Can the script-exporter or the script that runs the script exporter cache the result for up to 10m? This way we could run the script-exporter check every minute, but the script would only re-run every 10m.

**Comment by nkinkade** [comment]
This was resolved by m-lab/prometheus-script-exporter#6. Closing.

How I should note that apparently v2.0 does not actually fix this issue. The recommendation from the Prom devs, as noted above, is to try not to have a `scrape_interval` greater than 2 minutes.
